test/server/GraknRunnerBase.java

```diff
@@ -0,0 +1,186 @@
+package grakn.common.test.server;
+
+import org.apache.commons.io.FileUtils;
+import org.zeroturnaround.exec.ProcessExecutor;
+import org.zeroturnaround.exec.StartedProcess;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public abstract class GraknRunnerBase implements GraknRunner {
+    private static final String[] ARGS = System.getProperty("sun.java.command").split(" ");
+    private static final File DISTRIBUTION_FILE = ARGS.length > 1 ? new File(ARGS[1]) : null;
+
+    private static final String TAR = ".tar.gz";
+    private static final String ZIP = ".zip";
+
+    protected final String name;
+    private final File GRAKN_DISTRIBUTION_FILE;
+    private final Path GRAKN_TARGET_DIRECTORY;
+    private final String GRAKN_DISTRIBUTION_FORMAT;
+    protected final int port = ThreadLocalRandom.current().nextInt(40000, 60000);
+    protected final Path tmpDir;
+    protected final boolean debug;
+
+    private ProcessExecutor executor;
+    private StartedProcess graknProcess;
+
+    public GraknRunnerBase(String name) throws InterruptedException, TimeoutException, IOException {
+        this(name, DISTRIBUTION_FILE, false);
```

Comment:
Is it possible in Java to refactor `name` into an abstract method `name()` that we override in `GraknCoreRunner` and `GraknClusterRunner` to the appropriate values? (Not sure if we can use instance methods inside a `this` or `super` call, but if we can, it'd be cleaner)

Reply:
Fixed in 7330d27baf5d2d8ab8aa8859f4144c0a3fbf1c66